libraries/chain/steem_evaluator.cpp

```diff
@@ -944,7 +936,12 @@ void comment_evaluator::do_apply( const comment_operation& o )
       {
          com.last_update   = _db.head_block_time();
          com.active        = com.last_update;
-         strcmp_equal equal;
+         std::function< bool( const shared_string& a, const string& b ) > equal;
+
```

Comment:
Add an FC_TODO to test if things aren't broken after HF 21 so we can simplify this.